// angular2-toaster/src/toaster.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ToastComponent } from './toast.component';
import { ToasterContainerComponent } from './toaster-container.component';
import { ToasterService } from './toaster.service';
var ToasterModule = /** @class */ (function () {
    function ToasterModule() {
    }
    ToasterModule_1 = ToasterModule;
    ToasterModule.forRoot = function () {
        return {
            ngModule: ToasterModule_1,
            providers: [ToasterService, ToasterContainerComponent]
        };
    };
    ToasterModule.forChild = function () {
        return {
            ngModule: ToasterModule_1,
            providers: [ToasterContainerComponent]
        };
    };
    var ToasterModule_1;
    ToasterModule = ToasterModule_1 = __decorate([
        NgModule({
            imports: [CommonModule],
            declarations: [
                ToastComponent,
                ToasterContainerComponent
            ],
            exports: [
                ToasterContainerComponent,
                ToastComponent
            ]
        })
    ], ToasterModule);
    return ToasterModule;
}());
export { ToasterModule };
